views/hoc/AuthenticatedRoute.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useContext, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { AuthContext } from '../contexts/AuthContext'
+import { UserModel } from '../models'
+
+const AuthenticatedRoute = ({ children }) => {
+  const navigate = useNavigate()
+  const [user, setUser] = useContext(AuthContext)
+  const userId = window.sessionStorage.getItem('userId')
```

Comment:
I was thinking about this some more and was wondering if the API can do the same session authentication that it was doing with server rendered pages to get the current user. Like if we made an API call to `/api/v1/me` and it returned the current logged in user or `null`, can the API get the info it needs on the user from the request session?